### PR TITLE
Module4

### PR DESCRIPTION
**Pull request by BatiushkovNikita**

### CODE REVIEW

File: middlewares/cookie-parser.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+export default (req, res, next) => {
```
Comment (vinfinit) [inline middlewares/cookie-parser.js]: https://www.npmjs.com/package/cookie-parser

File: utils/streams.js [hunk 2]
```diff
@@ -72,19 +72,21 @@ function cssBundler(filePath) {
     readdir(filePath)
         .then(files => {
             files.map(fileName => {
```
Comment (vinfinit) [inline utils/streams.js]:
```javascript
.then(files => {
  let promiseList = files.map(fileName => {
    return appendToResult(filePath + '/' + fileName)
  });
  return promiseList
})
.then(() => appendToResult(appendixPath))
```

File: middlewares/cookie-parser.js (hunk 1 above)
Comment (BatiushkovNikita) [reply middlewares/cookie-parser.js]: yea, I know, but according to the task I thought that we need to implement our own parser

File: utils/streams.js (hunk 2 above)
Comment (BatiushkovNikita) [reply utils/streams.js]: fixed

File: app.js [hunk 3]
```diff
@@ -1,13 +1,36 @@
...
+import cookieParser from './middlewares/cookie-parser';
+import queryParser from './middlewares/query-parser';
+import productRouter from './routes/product';
+import userRouter from './routes/user';
+
+
+export default class App {
+
+    listen(port, cb) {
+        app.listen(port, cb);
```
[the remark this answers is not in the document]
Comment (BatiushkovNikita) [reply app.js]: fixed

File: routes/product.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        console.log('Returm ALL products');
+        res.json(productsData);
+    })
+    .post((req, res, next) => {
+        console.log('Add NEW product and return it');
+        res.json(req.body);
+    });
+
+router.param('id', (req, res, next, id) => {
+    let product = productsData.find(function (o) {
```
[the remark this answers is not in the document]
Comment (BatiushkovNikita) [reply routes/product.js]: fixed

**Pull request merged**